uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -0,0 +1,138 @@
+define([
+  'mocks/data/leave-request-data',
+  'mocks/apis/leave-request-api-mock',
+  'leave-absences/shared/models/instances/leave-request-instance',
+  'leave-absences/shared/modules/models',
+], function (mockData) {
+  'use strict';
+
+  describe('LeaveRequestInstance', function () {
+    var $provide,
+      LeaveRequestInstance,
+      LeaveRequestAPI,
+      $q,
+      OptionGroup,
+      $rootScope;
+
+    beforeEach(module('leave-absences.models', 'leave-absences.models.instances', 'leave-absences.mocks',
+      function (_$provide_) {
+        $provide = _$provide_;
+      }));
+
+    beforeEach(inject(function (_LeaveRequestAPIMock_) {
+      //LeaveRequestAPI is internally used by Model and hence need to be mocked
+      $provide.value('LeaveRequestAPI', _LeaveRequestAPIMock_);
+    }));
+
+    beforeEach(inject([
+      "LeaveRequestInstance",
+      "LeaveRequestAPI",
+      "$rootScope",
+      "$q",
+      'api.optionGroup',
+      function (_LeaveRequestInstance_, _LeaveRequestAPI_, _$rootScope_, _$q_, _OptionGroup_) {
+        LeaveRequestInstance = _LeaveRequestInstance_;
+        LeaveRequestAPI = _LeaveRequestAPI_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        OptionGroup = _OptionGroup_;
+      }]));
+
+    describe('cancel()', function () {
+      var optionGroupDeferred,
+        leaveRequestDeferred,
+        mockOptionValue,
+        mockUpdateResponse;
+
+      function commonSetup(returnData) {
+        optionGroupDeferred = $q.defer();
+        leaveRequestDeferred = $q.defer();
+        mockOptionValue = [{
+          name: "cancelled",
+          value: "1"
+        }];
+        mockUpdateResponse = returnData;
+
+        spyOn(OptionGroup, 'valuesOf').and.returnValue(optionGroupDeferred.promise);
+        spyOn(LeaveRequestInstance, 'update').and.returnValue(leaveRequestDeferred.promise);
+
+        optionGroupDeferred.resolve(mockOptionValue);
+        leaveRequestDeferred.resolve(mockUpdateResponse);
+      }
+
+      describe("success", function () {
+
+        beforeEach(function () {
+          commonSetup(mockData.singleDataSuccess());
+        });
+
+        it('updates the status_id of the instance', function () {
```

Comment:
You are testing the result of an async method (`cancel()`), so the `expect`s should be in the `then` function
I saw from dhaval's code that putting `$rootScope.$apply()` in the `afterEach` block should correctly allow you to test the expectation. If that doesn't work, you can run `$apply` directly after the `then()` function (see appraisals tests)
```js
var p;

beforeEach(function () {
  commonSetup(mockData.singleDataSuccess());

  LeaveRequestInstance.status_id = jasmine.any(String);
  p = LeaveRequestInstance.cancel()
});

afterEach(function () {
  $rootScope.$apply();
});

it('updates the status_id of the instance', function () {
  p.then(function () {
    expect(LeaveRequestInstance.status_id).toBe(mockUpdateResponse.values[0].status_id);
  })
});

it('OptionGroup.valuesOf gets called', function () {
  p.then(function () {
    expect(OptionGroup.valuesOf).toHaveBeenCalledWith('hrleaveandabsences_leave_request_status');
  });
});

it('LeaveRequestInstance.update gets called', function () {
  p.then(function () {
    expect(LeaveRequestInstance.update).toHaveBeenCalledWith({
      'status_id': mockOptionValue[0].value
    });
  });
});
```